api/agent/drivers/docker/docker.go

```diff
@@ -102,13 +164,41 @@ func NewDocker(conf drivers.Config) *DockerDriver {
 		}
 	}
 
+	liopts := docker.ListImagesOptions{All: false}
+	requiredImages := ListOfImageIDs(conf.RequiredImages)
+	liopts.Context = context.Background()
+
 	if conf.DockerLoadFile != "" {
 		err = loadDockerImages(driver, conf.DockerLoadFile)
 		if err != nil {
 			logrus.WithError(err).Fatalf("cannot load docker images in %s", conf.DockerLoadFile)
 		}
 	}
 
+	if conf.MaxImageCacheSize != 0 {
+		driver.imageCache = NewCache()
+
+		go func(context context.Context) {
+			liopts := docker.ListImagesOptions{All: false}
+			liopts.Context = context
+			images, err := driver.docker.ListImages(liopts)
+
+			if err != nil {
+				logrus.WithError(err).Fatalf("cannot list docker images %s", err)
```

Comment:
This is tricky... Maybe log and retry this? This step could be part of NewImageCleaner() too, it could retry for a while? If we crash here, we might crash in the middle of processing functions.

Reply:
would be nice to fatalf in the same goroutine as NewDocker if we're going to do it (and I agree that we should in this case probably, it probably means docker is dead or misconfigured)

Reply:
There is already a in the ListImages function, should we retry the retries?